state-chain/pallets/cf-swapping/src/tests.rs

```diff
@@ -2498,3 +2497,124 @@ fn swap_output_amounts_correctly_account_for_fees() {
 		});
 	}
 }
+
+#[test]
+fn test_buy_back_flip() {
+	new_test_ext().execute_with(|| {
+		const INTERVAL: BlockNumberFor<Test> = 5;
+		const SWAP_AMOUNT: AssetAmount = 1000;
+		const NETWORK_FEE: Permill = Permill::from_percent(2);
+
+		NetworkFee::set(NETWORK_FEE);
+
+		// The default buy interval is zero, and this means we don't buy back.
+		assert_eq!(FlipBuyInterval::<Test>::get(), 0);
```

Comment:
What is the purpose of this line? If you want to test the behaviour when interval = 0, you need to  ensure the relavant code gets called (calling on_initialize etc). Otherwise you might as well remove this.

Reply:
I think the purpose was to just check that it is set to 0 by default at genesis. Not very important, but worth leaving in i think.

Reply:
I think it is worth testing the behaviour when set to 0 in that case. It is important that we have logic for handling this separately, i.e. we don't divide by 0, for example.